DatadogSessionReplay/Sources/Recorder/RecordingCoordinator.swift

```diff
@@ -115,46 +113,68 @@ internal class RecordingCoordinator {
         srContextPublisher.setHasReplay(hasReplay)
     }
 
-    /// Captures the next recording if conditions are met.
-    private func captureNextRecord() {
-        /// We don't capture any snapshots if the RUM context has no view ID.
-        guard let rumContext = currentRUMContext,
-              let viewID = rumContext.viewID else {
-            return
-        }
+    private var lastFrameTimestamp: Date?
+    private var shouldSkipFrame: Bool {
+        return Date().timeIntervalSince(lastFrameTimestamp ?? .distantPast) < Constants.throttingRate
```

Comment:
suggestion/ Using `dateProvider` would let us add coverage to skipping frames without blocking tests thread. No risk of flakiness also as `dateProvider` mock is more explicit than "current real time".